lib/ohai/plugins/vmware.rb

```diff
@@ -0,0 +1,77 @@
+#
+# Author:: "Dan Robinson" <drobinson@getchef.com>
+# Author:: "Christopher M. Luciano" <cmlucian@us.ibm.com>
+# Copyright:: Copyright (c) 2014 Chef Software, Inc.
+# Copyright (C) 2015 IBM Corp.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Provides a set of attributes for a VMware ESX virtual machine with results
+# obtained from vmware-toolbox-cmd.  VMware Tools must be installed
+# on the virtual machine.
+#
+# Modify the path to vmware-toolbox-cmd in the call to get_vm_attributes for
+# your particular operating system and configuration
+#
+# Example:
+#
+# get_vm_attributes("/usr/bin/vmware-toolbox-cmd")
+#
+
+Ohai.plugin(:Vmware) do
+  provides "vmware"
+
+  def from_cmd(cmd)
+    so = shell_out(cmd)
+    so.stdout.split($/)[0]
+  end
+
+  def get_vm_attributes(vmtools_path)
+    if !File.exist?(vmtools_path)
+      Ohai::Log.debug("#{vmtools_path} not found")
+    else
+      vmware Mash.new
+      begin
+        # vmware-toolbox-cmd stat <param> commands
+        # Iterate through each parameter supported by the "vwware-toolbox-cmd stat" command, assign value
+        # to attribute "vmware[:<parameter>]"
+        [ "hosttime", "speed", "sessionid", "balloon", "swap", "memlimit", "memres", "cpures", "cpulimit" ].each do |param|
+          vmware[param.to_sym] = from_cmd("#{vmtools_path} stat #{param}")
+          if vmware[param.to_sym] =~ /UpdateInfo failed/
+            vmware[param.to_sym] = nil
+          end
+        end
+        # vmware-toolbox-cmd <param> status commands
+        # Iterate through each parameter supported by the "vwware-toolbox-cmd status" command, assign value
+        # to attribute "vmware[:<parameter>]"
+        [ "upgrade", "timesync" ].each do |param|
+          vmware[param.to_sym] = from_cmd("#{vmtools_path} #{param} status")
+        end
+      rescue
+        Ohai::Log.debug("Error while collecting VMware guest attributes")
+      end
+    end
+  end
+
+  collect_data(:default) do
+    get_vm_attributes("/usr/bin/vmware-toolbox-cmd")
+  end
+
+  collect_data(:linux) do
+    get_vm_attributes("/usr/bin/vmware-toolbox-cmd")
```

Comment:
If you drop `:default` then this plugin will only run on Linux.

If `/usr/bin/vmware-toolbox-cmd` is ubiquitous across most platforms (linux, osx, bsd), then drop the `collect_data(:linux)` block in favor of the `:default` one. You can add a `collect_data(:windows)` block later.

If you don't want this data collected on Windows right now and you use the `collect_data(:default)` block, you could do one of the following:
- Add a dependency on `"platform_family"` and don't call `get_vm_attributes(...)` if the [platform family is windows](https://github.com/chef/ohai/blob/master/lib/ohai/plugins/windows/platform.rb#L31)
- Check `RUBY_PLATFORM` for windows, [like this](https://github.com/chef/ohai/blob/master/spec/support/platform_helpers.rb#L9-L11)